infrastructure/src/main/java/org/corfudb/infrastructure/log/Compression/ZSTDCompression.java

```diff
@@ -0,0 +1,67 @@
+package org.corfudb.infrastructure.log.Compression;
+
+import com.github.luben.zstd.Zstd;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ZSTDCompression implements Codec {
+
+    private static final int DEFAULT_COMPRESSION_LEVEL = 3;
+
+    private static ZSTDCompression INSTANCE = new ZSTDCompression();
+
+    public static ZSTDCompression getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public ByteBuffer compress(ByteBuffer uncompressed) {
+        checkNotNull(uncompressed);
+        checkArgument(uncompressed.hasRemaining());
+
+        final int decompressedLength = uncompressed.remaining();
+        final int maxCompressedLength = (int) Zstd.compressBound(decompressedLength);
+
+        //TODO(Maithem): remove unpooled direct buffer allocations
```

Comment:
please get rid of TODO.
P.S. never leave todos in code

Reply:
Bet.

Reply:
It's not. It is already proven that using TODO is a bad practice that never works. You can see that check even in IntelliJ IDEA (it's enabled by default), it checks that there are no todos in the change.

What I can find on the internet for a couple of minutes http://wiki.c2.com/?TodoCommentsConsideredHarmful
For the exhaustive explanation/motivation read "Robert Martin clean code"

Reply:
I believe I was misunderstood. Bet means for sure. Thanks for explanation.

Reply:
Ok/